integration-tests/smoke/ccip/ccip_token_transfer_test.go

```diff
@@ -0,0 +1,266 @@
+)
+
+func TestTokenTransfer(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	config := &changeset.TestConfigs{}
+	tenv, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr, config)
+	inMemoryEnv := false
+
+	// use this if you are testing locally in memory
+	// tenv := changeset.NewMemoryEnvironmentWithJobsAndContracts(t, lggr, 2, 4, config)
+	// inMemoryEnv := true
+
+	e := tenv.Env
+	state, err := changeset.LoadOnchainState(e)
+	require.NoError(t, err)
+
+	// Chain and account setup
+	allChainSelectors := maps.Keys(e.Chains)
+	sourceChain, destChain := allChainSelectors[0], allChainSelectors[1]
+	ownerSourceChain := e.Chains[sourceChain].DeployerKey
+	ownerDestChain := e.Chains[destChain].DeployerKey
+
+	// Deploy and fund self-serve actors
+	selfServeSrcTokenPoolDeployer := createAndFundSelfServeActor(t, ownerSourceChain, e.Chains[sourceChain], big.NewInt(1e18), inMemoryEnv)
+	selfServeDestTokenPoolDeployer := createAndFundSelfServeActor(t, ownerDestChain, e.Chains[destChain], big.NewInt(1e18), inMemoryEnv)
+
+	// Deploy tokens and pool by CCIP Owner
+	srcToken, _, destToken, _, err := changeset.DeployTransferableToken(
+		lggr,
+		tenv.Env.Chains,
+		sourceChain,
+		destChain,
+		ownerSourceChain,
+		ownerDestChain,
+		state,
+		e.ExistingAddresses,
+		"OWNER_TOKEN",
+	)
+	require.NoError(t, err)
+
+	// Deploy Self Serve tokens and pool
+	selfServeSrcToken, _, selfServeDestToken, _, err := changeset.DeployTransferableToken(
+		lggr,
+		tenv.Env.Chains,
+		sourceChain,
+		destChain,
+		selfServeSrcTokenPoolDeployer,
+		selfServeDestTokenPoolDeployer,
+		state,
+		e.ExistingAddresses,
+		"SELF_SERVE_TOKEN",
+	)
+	require.NoError(t, err)
+
+	// Add all lanes.
+	require.NoError(t, changeset.AddLanesForAll(e, state))
+
+	// Mint and allow tokens for the router
+	changeset.MintAndAllow(t, e, state, map[uint64]*bind.TransactOpts{
+		sourceChain: ownerSourceChain,
+		destChain:   ownerDestChain,
+	}, map[uint64][]*burn_mint_erc677.BurnMintERC677{
+		sourceChain: {srcToken},
+		destChain:   {destToken},
+	})
+	changeset.MintAndAllow(t, e, state, map[uint64]*bind.TransactOpts{
+		sourceChain: selfServeSrcTokenPoolDeployer,
+		destChain:   selfServeDestTokenPoolDeployer,
+	}, map[uint64][]*burn_mint_erc677.BurnMintERC677{
+		sourceChain: {selfServeSrcToken},
+		destChain:   {selfServeDestToken},
+	})
+
+	tinyOneCoin := new(big.Int).SetUint64(1)
+
+	// Test scenarios are defined here
+	scenarios := []struct {
+		name                   string
+		srcChain               uint64
+		dstChain               uint64
+		tokenAmounts           []router.ClientEVMTokenAmount
+		receiver               common.Address
+		data                   []byte
+		expectedTokenBalances  map[common.Address]*big.Int
+		expectedExecutionState int
+	}{
+		{
+			name:     "Send token to EOA",
+			srcChain: sourceChain,
+			dstChain: destChain,
+			tokenAmounts: []router.ClientEVMTokenAmount{
+				{
+					Token:  srcToken.Address(),
+					Amount: tinyOneCoin,
```

Comment:
Yeah, should be just oneCoin. I think it was copypasted from USDC which required always to pass exactly 1 because this value is hardcoded in the mocked contracts